Updated compare columns dialog

HawardKetoyoMsatsi opened the pull request:
@shadrackkibet @africanmathsinitiative/developers @rdstern This is ready for review. 
This PR addresses issues suggested by @shadrackkibet in PR #6414 .

lloyddewit commented:
@HawardKetoyoMsatsi Thank you for the PR. When you open a PR, please could you add more details about what has changed. Is this a bug fix or enhancement? Is it related to an specific GitHub issue? 

@N-thony Please could you peer review? 

Thanks

N-thony commented:
> @HawardKetoyoMsatsi Thank you for the PR. When you open a PR, please could you add more details about what has changed. Is this a bug fix or enhancement? Is it related to an specific GitHub issue?
> 
> @N-thony Please could you peer review?
> 
> Thanks

@lloyddewit I'm happy to do it.

N-thony commented on instat/dlgCompareColumns.vb:
```diff
@@ -20,13 +20,17 @@ Public Class dlgCompareColumns
-    Private clsCompareColumns, clsIfElseCompareFunction, clsAbsoluteFunction As New RFunction
+    Private iDialogHeight As Integer
+    Private iBaseMaxY
+    Private clsCompareColumns, clsAbsoluteFunction As New RFunction
```
Rename clsCompareColumnsFunction? just for consistency.

N-thony commented on instat/dlgCompareColumns.vb:
```diff
@@ -47,10 +51,12 @@ Public Class dlgCompareColumns
 
-        ucrPnlOptions.AddFunctionNamesCondition(rdoByRow, "ifelse")
-        ucrPnlOptions.AddFunctionNamesCondition(rdoByValue, {"%in%", "compare_columns"})
+        'ucrPnlOptions.AddParameterIsROperatorCondition(rdoByRow, "==")
+        'ucrPnlOptions.AddParameterIsROperatorCondition(rdoByRow, "<=")
+        'ucrPnlOptions.AddFunctionNamesCondition(rdoByValue, {"compare_columns"})
```
Remove or comment if need in the future?

N-thony commented on instat/dlgCompareColumns.vb:
```diff
@@ -173,7 +170,7 @@ Public Class dlgCompareColumns
         ucrReceiverSecond.AddAdditionalCodeParameterPair(clsSubtractOperator, New RParameter("second", bNewIncludeArgumentName:=False, iNewPosition:=1), iAdditionalPairNo:=3)
 
-        ucrPnlOptions.SetRCode(ucrBase.clsRsyntax.clsBaseFunction, bReset)
+        ' ucrPnlOptions.SetRCode(, bReset)
```
Remove or comment if need in the future all commented lines?

N-thony commented on instat/dlgCompareColumns.vb:
```diff
@@ -225,54 +213,75 @@ Public Class dlgCompareColumns
     Private Sub ucrPnlOptions_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrPnlOptions.ControlValueChanged
         If rdoByValue.Checked Then
-            grpComparisions.Visible = True
+            'grpComparisions.Visible = True
```
Again

rdstern reviewed (approved):
Looks ok to merge to me!

HawardKetoyoMsatsi commented:
@N-thony  I have fixed the issue raised concerning unnecessary comments.
@lloyddewit this PR only addresses issues suggested by @shadrackkibet in PR #6414

Patowhiz commented:
@HawardKetoyoMsatsi in this PR does this dialog expand and shrink it's size based on the radio button checked? If do, then the only thing left is linking the receiver to the `ucrSave` control as outlined by @rdstern in issue #6397. For this dialog, I think this should just be 1 line as done in most dialogs in PR #6434, we can discuss about it further.

Patowhiz commented on instat/dlgCompareColumns.vb:
```diff
@@ -225,54 +206,66 @@ Public Class dlgCompareColumns
-        Else
-            grpComparisions.Visible = False
-            ucrBase.clsRsyntax.SetBaseRFunction(clsIfElseCompareFunction)
+            Me.Size = New System.Drawing.Size(Me.Width, iDialogHeight)
```
Why not just use `Me.height`, since only the height is being changed all the time?

HawardKetoyoMsatsi replied:
@Patowhiz I actually can't figure out how to use only Me.height that's why I used Me.size, could you please edit the code then I can try and accept your suggestion from my end.

Patowhiz replied:
Okay. Will make a PR to your branch.

Patowhiz commented:
@HawardKetoyoMsatsi in regards to my [comment ](https://github.com/africanmathsinitiative/R-Instat/pull/6433#issuecomment-832758004)above, ignore it. After looking at your code the receiver is already linked and working correctly.

HawardKetoyoMsatsi commented on instat/dlgCompareColumns.vb:
```diff
@@ -20,13 +20,17 @@ Public Class dlgCompareColumns
-    Private clsCompareColumns, clsIfElseCompareFunction, clsAbsoluteFunction As New RFunction
+    Private iDialogHeight As Integer
+    Private iBaseMaxY
+    Private clsCompareColumns, clsAbsoluteFunction As New RFunction
```
@N-thony am done renaming

lloyddewit reviewed (changes requested):
Thanks, this looks good, just a few small suggestions

HawardKetoyoMsatsi commented on instat/dlgCompareColumns.vb:
```diff
@@ -108,108 +110,87 @@ Public Class dlgCompareColumns
-            Else
+        If ucrReceiverFirst.IsEmpty OrElse ucrReceiverSecond.IsEmpty OrElse Not ucrSaveLogical.IsComplete() Then
+            ucrBase.OKEnabled(False)
+        Else
```
No, there is no ElseIf in Sub TestOkEnabled()

HawardKetoyoMsatsi commented on instat/dlgCompareColumns.vb:
```diff
@@ -225,54 +206,66 @@ Public Class dlgCompareColumns
-                If {"integer", "numeric"}.Contains(ucrReceiverFirst.strCurrDataType) AndAlso {"integer", "numeric"}.Contains(ucrReceiverSecond.strCurrDataType) Then
+        If Not ucrReceiverSecond.IsEmpty AndAlso Not ucrReceiverFirst.IsEmpty Then
+            If {"integer", "numeric"}.Contains(ucrReceiverFirst.strCurrDataType) AndAlso {"integer", "numeric"}.Contains(ucrReceiverSecond.strCurrDataType) Then
+                If rdoByRow.Checked Then
```
I have brought back If rdoByRow.Checked since the condition checking for datatype applies to both rdoByRow and rdoByValue, The check for rdoByRow should thus come inside the check for the datatype. if I run with your suggestion the check doesnt work when rdoByValue is checked

lloyddewit commented on instat/dlgCompareColumns.vb:
```diff
@@ -225,54 +206,66 @@ Public Class dlgCompareColumns
-            ucrBase.clsRsyntax.SetBaseRFunction(clsIfElseCompareFunction)
+            Me.Size = New System.Drawing.Size(Me.Width, iDialogHeight)
+            ucrBase.Location = New Point(ucrBase.Location.X, iBaseMaxY)
+            ucrSaveLogical.Location = New Point(ucrSaveLogical.Location.X, iBaseMaxY / 1.07)
```
@HawardKetoyoMsatsi You marked this as resolved but the code was not changed. Please could you clarify?

lloyddewit reviewed (approved):
Thanks! There was just one very small comment so I approved

lloyddewit commented:
@rdstern please could you retest/approve? thanks

lloyddewit commented:
@shadrackkibet I'm happy to merge, but if you'd like to review first, then please go ahead

shadrackkibet commented:
@lloyddewit looks good.

dannyparsons commented:
@HawardKetoyoMsatsi A more descriptive title for the PR would be useful in future as these appear in the release notes. "Updated" is a bit too vague.
